OWDTestToolkit/apps/settings.py

```diff
@@ -711,6 +720,78 @@ def set_time_to_now(self):
         time.sleep(2)
         self.marionette.switch_to_frame()
 
+    def enable_sim_security(self, enable, pin, puk=None):
+        """Enable SIM PIN security, if required"""
+
+        sim_manager = self.UTILS.element.getElement(DOM.Settings.sim_manager, "SIM Manager menu")
+        self.UTILS.element.simulateClick(sim_manager)
+        time.sleep(3)
+
+        sim_security = self.UTILS.element.getElement(DOM.Settings.sim_manager_sim_security, "SIM Security menu")
+        sim_security.tap()
+
+        switcher = self.UTILS.element.getElement(DOM.Settings.dual_sim_switch_pin_sim1, "SIM PIN switch")
+        switch_input = self.marionette.find_element('css selector', 'input', switcher.id)
+        enabled = switch_input.get_attribute("checked") is not None
+        self.UTILS.reporting.info("PIN enabled: {}".format(enabled is not None))
+
+        # If PIN is already in the correct state, just return
+        if enabled and not enable or (not enabled and enable):
+            self.UTILS.reporting.info("Pin not in correct state. Current: {} Requested: {}. Changing.".
+                                      format(enabled, enable))
+            # First of all, verify if the SIM is locked, and, in that case, unlock using PUK
+            switcher.tap()
+            time.sleep(1)
+            unlocked = self.unlock_sim(puk, pin)
+            if unlocked:
+                self.check_security_menu()
```

Comment:
This is the trick for the glitch after pressing Done button, just moved to a method